components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/ext/listener/ExtensionListenerUtil.java

```diff
@@ -264,10 +264,18 @@ private static boolean handleExtensionResponse(MessageContext messageContext,
         }
         // handle payload
         processResponsePayload(extensionResponseDTO, messageContext);
+        //Extract customProperty from DTO
+        Map<String, Object> customProperty = extensionResponseDTO.getCustomProperty();
+        //set Additional analytics Properties to the message context.
+        if(customProperty != null && customProperty.get(APIMgtGatewayConstants.ADDITIONAL_ANALYTICS_PROPS) != null ) {
+            Map<String, Object> customElkAttributes =
+                    (Map<String, Object>) customProperty.get(APIMgtGatewayConstants.ADDITIONAL_ANALYTICS_PROPS);
+            messageContext.setProperty(APIMgtGatewayConstants.ADDITIONAL_ANALYTICS_PROPS, customElkAttributes);
```

Comment:
Do we really need to do the casting to `Map<String, Object>` here as we are just setting it back to the `messageContext` as an object?